SpatialGDK/Source/SpatialGDK/Private/Tests/LoadBalanceEnforcerTest.cpp

```diff
@@ -0,0 +1,358 @@
+#include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Schema/AuthorityIntent.h"
+#include "Tests/TestingSchemaHelpers.h"
+
+#include "CoreMinimal.h"
+
+#include "Schema/ClientEndpoint.h"
+#include "SpatialConstants.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/EntityComponentTypes.h"
+#include "SpatialView/EntityView.h"
+#include "SpatialView/ViewDelta.h"
+#include "Tests/SpatialView/SpatialViewUtils.h"
+#include "Utils/SchemaOption.h"
+
+#define LOADBALANCEENFORCER_TEST(TestName) GDK_TEST(Core, SpatialLoadBalanceEnforcer, TestName)
+
+// Test Globals
+namespace
+{
+const PhysicalWorkerName UnrealWorker = TEXT("UnrealWorker");
+const WorkerRequirementSet UnrealRequirementSet = WorkerRequirementSet{ WorkerRequirementSet{ WorkerAttributeSet{ UnrealWorker } } };
+
+const PhysicalWorkerName ThisWorker = TEXT("ThisWorker");
+const PhysicalWorkerName OtherWorker = TEXT("OtherWorker");
+const PhysicalWorkerName ClientWorker = TEXT("ClientWorker");
+const PhysicalWorkerName OtherClientWorker = TEXT("OtherClientWorker");
+
+const WorkerRequirementSet ThisRequirementSet =
+	WorkerRequirementSet{ WorkerRequirementSet{ WorkerAttributeSet{ FString::Printf(TEXT("workerId:%s"), *ThisWorker) } } };
+const WorkerRequirementSet OtherRequirementSet =
+	WorkerRequirementSet{ WorkerRequirementSet{ WorkerAttributeSet{ FString::Printf(TEXT("workerId:%s"), *OtherWorker) } } };
+const WorkerRequirementSet OtherClientRequirementSet =
+	WorkerRequirementSet{ WorkerRequirementSet{ WorkerAttributeSet{ FString::Printf(TEXT("workerId:%s"), *OtherClientWorker) } } };
+
+constexpr VirtualWorkerId ThisVirtualWorker = 1;
+constexpr VirtualWorkerId OtherVirtualWorker = 2;
+
+constexpr Worker_EntityId EntityIdOne = 1;
+
+constexpr Worker_ComponentId TestComponentIdOne = 123;
+constexpr Worker_ComponentId TestComponentIdTwo = 456;
+
+const TArray<Worker_ComponentId> PresentComponents = {
+	SpatialConstants::ENTITY_ACL_COMPONENT_ID,		   SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID,
+	SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID, SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID,
+	SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID,	   SpatialConstants::HEARTBEAT_COMPONENT_ID
+};
+const TArray<Worker_ComponentId> NonAclLBComponents = { SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID,
+														SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID,
+														SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID };
+const TArray<Worker_ComponentId> TestComponentIds = { TestComponentIdOne, TestComponentIdTwo };
+const TArray<Worker_ComponentId> ClientComponentIds = { SpatialConstants::HEARTBEAT_COMPONENT_ID,
+														SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID };
+
+TUniquePtr<SpatialVirtualWorkerTranslator> CreateVirtualWorkerTranslator()
+{
+	TUniquePtr<SpatialVirtualWorkerTranslator> VirtualWorkerTranslator = MakeUnique<SpatialVirtualWorkerTranslator>(nullptr, ThisWorker);
```

Comment:
This looks like a sketchy nullptr. However, the alternative is using a UObject here, and that causes all sorts of issues, in a class which really has little to do with unreal. The way the LB enforcer uses the virtual worker translator will never cause this to be an issue, and we're going to rip this all out for strategy worker anyway. So for my own sanity, just in these tests, this is a nullptr.